File: BookShop.BusinessLogic/Models/PrintedEditionModel.cs
﻿using BookShop.DataAccess.Entities;
using static BookShop.DataAccess.Entities.Enums.Enums.EntityFields;

namespace BookShop.BusinessLogic.Models
{
    public class PrintedEditionModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public bool IsRemoved { get; set; }
        public Status Status { get; set; }
        public Currency Currency { get; set; }
        public PrintedEditionType Type { get; set; }

        public PrintedEditionModel(PrintedEdition printedEdition)
        {
            Id = printedEdition.Id;
            Name = printedEdition.Name;
            Description = printedEdition.Description;
            Price = printedEdition.Price;
            IsRemoved = printedEdition.IsRemoved;
            Status = printedEdition.Status;
            Currency = printedEdition.Currency;
            Type = printedEdition.Type;
        }
    }
}

File: BookShop.ClientApp/src/app/models/printedEditionModel.ts
import { PrintedEditionType } from '../enums/printedEditionType';
import { Currency } from '../enums/currency';
import { AuthorModel } from 'src/app/models/authorModel';

export class PrintedEditionModel {
    public id: number;
    public name: string;
    public description: string;
    public price: number;
    public type: PrintedEditionType;
    public currency: Currency;
    public authorModels: AuthorModel[];

    constructor() {

        this.type = PrintedEditionType.None as number;
        this.currency= Currency.None as number;
        this.authorModels=[];
    }
}